web/pgadmin/browser/server_groups/servers/databases/schemas/tables/indexes/static/js/index.ui.js

```diff
@@ -353,10 +356,94 @@ export default class IndexSchema extends BaseUISchema {
         },
         node:'column',
       },{
-        id: 'fillfactor', label: gettext('Fill factor'), cell: 'string',
-        type: 'int', disabled: () => inSchema(indexSchemaObj.node_info),
+        id: 'fillfactor', label: gettext('Fill factor'), deps: ['amname'], cell: 'string',
+        type: 'int', disabled: (state) => {
+          return !_.includes(['btree', 'hash', 'gist', 'spgist'], state.amname) || inSchema(indexSchemaObj.node_info);
+        },
         mode: ['create', 'edit', 'properties'],
         min: 10, max:100, group: gettext('Definition'),
+        depChange: state => {
+          if (!_.includes(['btree', 'hash', 'gist', 'spgist'], state.amname)) {
+            return {fillfactor: ''};
+          }
+        }
+      },{
+        id: 'gin_pending_list_limit', label: gettext('Gin pending list limit'), cell: 'string',
+        type: 'int', deps: ['amname'], disabled: state => state.amname !== 'gin' || inSchema(indexSchemaObj.node_info),
+        mode: ['create', 'edit', 'properties'],
+        group: gettext('Definition'),
+        depChange: state => {
+          if (state.amname !== 'gin') {
+            return {gin_pending_list_limit: ''};
+          }
+        }
+      },{
+        id: 'pages_per_range', label: gettext('Pages per range'), cell: 'string',
```

Comment:
Help String "help_str=gettext("Number of table blocks that make up one block range for each entry of a BRIN index").